torch/signal/windows/windows.py

```diff
@@ -0,0 +1,315 @@
+
+     Args:
+         function_name (str): name of the window function.
+         window_length (int): length of the window.
+         dtype (:class:`torch.dtype`): the desired data type of the window tensor.
+         layout (:class:`torch.layout`): the desired layout of the window tensor.
+     """
+
+    def is_floating_type(t: _dtype) -> bool:
+        return t == torch.float32 or t == torch.bfloat16 or t == torch.float64 or t == torch.float16
+
+    def is_complex_type(t: _dtype) -> bool:
+        return t == torch.complex64 or t == torch.complex128 or t == torch.complex32
+
+    if window_length < 0:
+        raise ValueError(f'{function_name} requires non-negative window_length, got window_length={window_length}')
+    if layout is not torch.strided:
+        raise ValueError(f'{function_name} is not implemented for sparse types, got: {layout}')
+    if not is_floating_type(dtype) and not is_complex_type(dtype):
+        raise ValueError(f'{function_name} expects floating point dtypes, got: {dtype}')
+
+
+@_add_docstr(
+    r"""
+Computes a window with an exponential waveform.
+Also known as Poisson window.
+
+The exponential window is defined as follows:
+
+.. math::
+    w(n) = \exp{\left(-\frac{|n - center|}{\tau}\right)}
+    """,
+    r"""
+
+Args:
+    window_length (int): the length of the output window.
+        In other words, the number of points of the ee window.
+    periodic (bool, optional): If `True`, returns a periodic window suitable for use in spectral analysis.
+        If `False`, returns a symmetric window suitable for use in filter design. Default: `True`.
+    center (float, optional): his value defines where the center of the window will be located.
+        In other words, at which sample the peak of the window can be found.
+        Default: `window_length / 2` if `periodic` is `True` (default), else `(window_length - 1) / 2`.
+    tau (float, optional): the decay value.
+        For `center = 0`, it's suggested to use :math:`\tau = -\frac{(M - 1)}{\ln(x)}`,
+        if `x` is the fraction of the window remaining at the end. Default: 1.0.
+    """ +
+    r"""
+
+.. note::
+    The window is normalized to 1 (maximum value is 1), however, the 1 doesn't appear if `M` is even
+    and `periodic` is `False`.
+
+Keyword args:
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples:
+    >>> # Generate an exponential window without keyword args.
+    >>> torch.signal.windows.exponential(10)
+    tensor([0.0067, 0.0183, 0.0498, 0.1353, 0.3679, 1.0000, 0.3679, 0.1353, 0.0498,
+    0.0183])
+
+    >>> # Generate a symmetric exponential window and decay factor equal to .5
+    >>> torch.signal.windows.exponential(10,periodic=False,tau=.5)
+    tensor([1.2341e-04, 9.1188e-04, 6.7379e-03, 4.9787e-02, 3.6788e-01, 3.6788e-01,
+    4.9787e-02, 6.7379e-03, 9.1188e-04, 1.2341e-04])
+    """.format(
+        **factory_common_args
+    ),
+)
+def exponential(window_length: int,
```

Comment:
In SciPy the signature is

```
exponential(M, center=none, tau=1.0, sym=True)
```

(https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.windows.exponential.html)

what are your thoughts on diverging the order and name of the parameters?

Reply:
Given our experience with other NumPy operations, I suggest being consistent unless there is a compelling reason to diverge.

cc @lezcano

Reply:
I agree that doing a rebranding for no good reason is not good. I think we should maintain SciPy's signature.

Reply:
Awesome. This decision helps a lot with OpInfo tests. I'll update the signatures.

Reply:
Done!